client/src/contexts/index.jsx

```diff
@@ -59,21 +67,35 @@ export function AuthProvider({ children }) {
     return addDoc(ridesCollectionReference, body);
   }
 
-  function getRides(params) {
-    return getDoc(ridesCollectionReference, params);
+  async function getRides(params) {
+    const rides = [];
+    const q = query(ridesCollectionReference, where('start', '==', params));
+    const querySnapshot = await getDocs(q);
+    querySnapshot.forEach((ride) => {
+      rides.push(ride.data());
+    });
+    return rides;
   }
 
   function addReview(body) {
     return addDoc(reviewsCollectionReference, body);
   }
 
-  function getReviews(params) {
-    return getDocs(reviewsCollectionReference, params);
+  async function getReviews(params) {
+    const reviews = [];
+    const reviewsSnapshot = await getDocs(reviewsCollectionReference, params);
+    reviewsSnapshot.forEach((review) => {
+      reviews.push(review.data());
+    });
+    return reviews;
   }
 
   function addMessage(body) {
     return addDoc(messagesCollectionReference, body);
   }
+  // function getMessages(params) {
+  //   return getDocs(messagesCollectionReference, params);
```

Comment:
I think you might still be working on this.

Reply:
deleting now!